crates/orderbook/src/database/orders.rs

```diff
@@ -254,6 +257,26 @@ impl OrderStoring for Postgres {
     }
 }
 
+#[async_trait]
+impl LimitOrderCounting for Postgres {
+    async fn count(&self, owner: H160) -> Result<u64> {
+        let _timer = super::Metrics::get()
+            .database_queries
+            .with_label_values(&["count_limit_orders"])
+            .start_timer();
+
+        let mut ex = self.pool.acquire().await?;
+        Ok(database::orders::count_limit_orders(
+            &mut ex,
+            now_in_epoch_seconds().try_into().unwrap(),
+            &ByteArray(owner.0),
+        )
+        .await?
+        .try_into()
+        .unwrap())
```

Comment:
I guess we can avoid using unwrap with this?

```suggestion
        database::orders::count_limit_orders(
            &mut ex,
            now_in_epoch_seconds().try_into().unwrap(),
            &ByteArray(owner.0),
        )
        .await?
        .try_into()
        .context("failed to count limit orders")
```

Reply:
This would only crash if the count of limit orders is negative AFAIU. I would also be loud in these cases, as it likely means that there is a logic error (and not just some invalid data).